### PR TITLE
fix(stream): Fix memory leak from allocators

### PR DESCRIPTION
**Pull request by jarifibrahim**
There were two memory leaks in the stream framework and this PR fixes both of them.
The reuse of allocators was causing the memory leak.

Fixes DGRAPH-2650

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1591)

### CODE REVIEW

File: stream.go [hunk 1]
```diff
@@ -206,6 +201,8 @@ func (st *Stream) produceKVs(ctx context.Context, threadId int) error {
 	}
 	defer txn.Discard()
 
+	outList := new(pb.KVList)
+	outList.AllocRef = st.newAllocator(threadId).Ref
```
Comment (NamanJain8) [inline stream.go:205]: Why do we want to move it outside iterate? I see no difference with case when if it is inside `iterate`.

**Review by NamanJain8 — APPROVED** [review]
Looks good. Just a minor comment.

File: stream.go (hunk 1 above)
Comment (NamanJain8) [reply stream.go:205]: Also, do we want a test where `KeyToList()` does not use the allocator?

Comment (jarifibrahim) [reply stream.go:205]: The outlist was inside `iterate` function and `sendit` creates a new outlist.
So if the following sequence of calls happen
```
iterate <- New outlist created (say o1)
sendIt <- Old outlist (o1) pushed to kvchan and new outlist (say o2) created 
iterate <- new outlist (o3 created)
```
Notice that we did not push 02 to the `kvChan` and so the allocator in `o2` will leak.
Moving the outlist outside prevents this problem.

Comment (jarifibrahim) [reply stream.go:205]: I'll push a test for this.

**Review by manishrjain — COMMENTED** [review]
Is this PR still relevant?

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1591#-:-MMqYjRv6kr253fDKtxE:b-5bnhf)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @manishrjain and @NamanJain8)

**Pull request closed without merging**